### PR TITLE
Get THOR Ledger address in settings

### LINKED ISSUE #1664: [Settings] Extend UI to add business logic to connect Ledger
- (1) UI changes #1657
- (2) Business logic #1655 

Combine (1) + (2)

### PR DESCRIPTION
**Pull request by veado**
Most of the logic is very similar to what we had developed for Ledger `BNB`/`BTC` before - but for THORChain now incl. some improvements:

- [x] Refactor types (e.g. moving `WalletBalance` / `WalletBalances` from `/types/wallet` to `/services/wallet/types`
- [x] Add logic to get Ledger address in `SettingsView`
- [x] Persistent state of Ledger address in memory (in `services/wallet/ledger`, `services/{chain}/ledger`) - incl. loading/error states
- [x] Refactor `useLedger`  to get data / callbacks from `WalletContext`
- [x] Remove `services/chain/ledger` in favour of `services/wallet/ledger`
- [x] Few tweaks for Settings UI

https://user-images.githubusercontent.com/61792675/129489042-74f8137f-641f-4b1d-b5ea-ff6b923f212c.mp4

Close #1664 
Part of #1570

### CODE REVIEW

File: src/renderer/views/wallet/SettingsView.tsx [hunk 1]
```diff
@@ -23,189 +22,183 @@ import { useEthereumContext } from '../../contexts/EthereumContext'
...
     O.getOrElse(() => '')
   )
 
-  const thorAccount$ = useMemo(
+  const thorLedgerAccount: WalletAddress = useMemo(
+    () => ({
+      type: 'ledger',
+      address: FP.pipe(
+        thorLedgerAddressRD,
+        RD.mapLeft((errorID) => Error(`Could not get THOR address from Ledger ${errorID}`))
```
Comment (veado) [inline src/renderer/views/wallet/SettingsView.tsx]: ^ Add `i18n`

File: src/renderer/hooks/useLedger.ts [hunk 2]
```diff
@@ -1,40 +1,38 @@
...
-      ).subscribe(setAddress)
-    },
-    [network, setAddress]
+  const address = useObservableState<LedgerAddressRD>(
+    FP.pipe(getLedgerAddressByChain$(chain), RxOp.shareReplay(1)),
+    RD.initial
   )
 
+  const askAddress = useCallback(() => {
+    const sub = askLedgerAddressByChain$(chain, network).subscribe()
```
Comment (veado) [inline src/renderer/hooks/useLedger.ts:32]: Add comment why subscription is needed here ^

File: src/renderer/components/wallet/settings/Settings.tsx [hunk 3]
```diff
@@ -97,52 +91,67 @@ export const Settings: React.FC<Props> = (props): JSX.Element => {
...
-                        )}
-                      </Styled.ChainContent>
-                    ))}
+                    <Styled.ChainName>{chain}</Styled.ChainName>
+                    {accounts.map((account, j) => {
+                      const { type } = account
+
+                      return (
+                        <Styled.ChainContent key={j}>
+                          <Styled.AccountPlaceholder>{type}</Styled.AccountPlaceholder>
```
Comment (veado) [inline src/renderer/components/wallet/settings/Settings.tsx]: Add `i18n` for type

File: src/renderer/i18n/en/ledger.ts [hunk 4]
```diff
@@ -1,14 +1,14 @@
...
-  'ledger.errors.already.in.use': 'The device is used in other app',
-  'ledger.errors.no.app': 'No ledger app opened. Please open appropriate app on the device.',
-  'ledger.errors.wrong.app': 'Opened other app on the device. Quit and open appropriate app.',
-  'ledger.errors.denied': 'You denied the request',
-  'ledger.errors.unknown': 'Unknown Error'
+  'ledger.error.nodevice': 'No device connected',
+  'ledger.error.inuse': 'The device is used in other app',
+  'ledger.error.noapp': 'No ledger app opened. Please open appropriate app on the device.',
+  'ledger.error.wrongapp': 'Opened other app on the device. Quit and open appropriate app.',
+  'ledger.error.denied': 'Usage of Ledger has bee denied',
```
Comment (gromxyz) [inline src/renderer/i18n/en/ledger.ts]: Typo: `bee` => `been`

File: src/renderer/services/wallet/ledger.ts [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+  switch (chain) {
+    case THORChain:
+      return THOR.ledgerAddress$
+    default:
+      throw Error(`Ledger has not been implemented for ${chain} yet`)
+  }
+}
+
+/**
+ * Get ledger address from memory
```
Comment (gromxyz) [inline src/renderer/services/wallet/ledger.ts]: Comments can be updated

**Review by gromxyz — APPROVED** [review]

**Pull request merged**